msbuild/Xamarin.MacDev.Tasks.Core/Tasks/DetectSigningIdentityTaskBase.cs

```diff
@@ -489,20 +494,30 @@ where p.DeveloperCertificates.Any (d => d.Thumbprint == c.Thumbprint)
 			int matchLength;
 
 			// find matching provisioning profiles with compatible appid, keeping only those with the longest matching (wildcard) ids
+			Log.LogMessage (MessageImportance.Low, "Finding matching provisioning profiles with compatible AppID, keeping only those with the longest matching (wildcard) IDs.");
 			foreach (var pair in pairs) {
 				var appid = ConstructValidAppId (pair.Profile, identity.BundleId, out matchLength);
-				if (appid != null && matchLength >= bestMatchLength) {
-					if (matchLength > bestMatchLength) {
-						bestMatchLength = matchLength;
-						matches.Clear ();
-					}
+				if (appid != null) {
+					if (matchLength >= bestMatchLength) {
+						if (matchLength > bestMatchLength) {
+							bestMatchLength = matchLength;
+							foreach (var previousMatch in matches)
+								Log.LogMessage (MessageImportance.Low, "AppID: {0} was ruled out because we found a better match: {1}.", previousMatch.AppId, appid);
+							matches.Clear ();
+						}
 
-					var match = identity.Clone ();
-					match.SigningKey = pair.SigningKey;
-					match.Profile = pair.Profile;
-					match.AppId = appid;
+						var match = identity.Clone ();
+						match.SigningKey = pair.SigningKey;
+						match.Profile = pair.Profile;
+						match.AppId = appid;
 
-					matches.Add (match);
+						matches.Add (match);
+					} else {
+						string currentMatches = "";
+						foreach (var match in matches)
+							currentMatches += $"{match}; ";
+						Log.LogMessage (MessageImportance.Low, "AppID: {0} was ruled out because we already found better matches: ", appid, currentMatches);
```

Comment:
@VincentDondain you're passing two arguments to format the string, but there's only `{0}` in the actual string, no `{1}`.

Reply:
Well played that was an easter egg and no one noticed it :trollface: 

Will fix

Reply:
Fixed https://github.com/xamarin/xamarin-macios/pull/3262